lib/graphQl/resolvers.js

```diff
@@ -0,0 +1,124 @@
+  path = parent.relationships[attribute].links.related;
+  return resolvers.rerouteTo("GET", path, { }, req, ast);
+};
+
+resolvers.create = function(resourceConfig, parent, args, req, ast) {
+  var path = jsonApi._apiConfig.pathPrefix + resourceConfig.resource;
+  var data = resolvers.generateResourceFromArgs(args, resourceConfig);
+  return resolvers.rerouteTo("POST", path, { data: data }, req, ast);
+};
+
+resolvers.update = function(resourceConfig, parent, args, req, ast) {
+  var path = jsonApi._apiConfig.pathPrefix + resourceConfig.resource + '/' + args.tags.id;
+  var data = resolvers.generateResourceFromArgs(args, resourceConfig);
+  return resolvers.rerouteTo("PATCH", path, { data: data }, req, ast);
+};
+
+resolvers.delete = function(resourceConfig, parent, args, req, ast) {
+  var path = jsonApi._apiConfig.pathPrefix + resourceConfig.resource + '/' + args.id;
+  var resource;
+  return resolvers.rerouteTo("GET", path, { }, req, ast)
+    .then(function(originalResource) {
+      resource = originalResource;
+      return resolvers.rerouteTo("DELETE", path, { }, req, ast);
+    }).then(function() {
+      return resource;
+    });
+};
+
+resolvers.rerouteTo = function(method, path, args, req, ast) {
+  return new Promise(function(resolve, reject) {
+    rerouter.route({
+      method: method,
+      uri: path,
+      params: _.assign({
+        fields: resolvers.generateFieldsQueryFromAst(ast),
+        filter: resolvers.generateFilterQueryFromAst(ast)
+      }, args),
+      originalRequest: {
+        headers: req.headers || { },
+        cookies: req.cookies || { }
+      }
+    }, function(err, json) {
+      if (err) return reject(err.errors.map(function(e) { return e.detail; }));
+      resolve(json.data);
+    });
+  });
+};
+
+resolvers.generateResourceFromArgs = function(args, resourceConfig) {
+  if ((Object.keys(args).length === 1) && (args[resourceConfig.resource])) {
+    args = args[resourceConfig.resource];
+  }
+
+  var data = {
+    type: resourceConfig.resource,
+    attributes: { },
+    relationships: { }
+  };
+
+  Object.keys(resourceConfig.attributes).forEach(function(attribute) {
+    var joiSchema = resourceConfig.attributes[attribute];
+    if (!args[attribute]) return;
+    if (!joiSchema._settings) {
+      data.attributes[attribute] = args[attribute];
+    } else {
+      data.relationships[attribute] = {
+        data: args[attribute]
+      };
+      [].concat(data.relationships[attribute].data).forEach(function(relation) {
+        relation.type = joiSchema._settings.__one || joiSchema._settings.__many;
+      })
+    }
+  });
+
+  return data;
+};
+
+resolvers.generateFieldsQueryFromAst = function(ast) {
+  var arrays = (ast.fieldASTs || []).map(function(fieldAST) {
+    return fieldAST.selectionSet.selections || [ ];
+  });
+  var combined = [].concat.apply([], arrays);
+  var fields = []
+  combined.forEach(function(thing) {
+    fields.push(thing.name.value);
+  });
```

Comment:
Yes for this one, no for the second one below - that one uses an object instead of an array.